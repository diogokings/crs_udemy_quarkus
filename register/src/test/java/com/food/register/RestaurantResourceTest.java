package com.food.register;

import static io.restassured.RestAssured.given;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(RegisterLifecycleManagerTest.class)
public class RestaurantResourceTest {

    @Test
    @DataSet("restaurant-find_all-success.yml")
    public void testFindAllRestaurants() {
    	String response = given()
          .when().get("/restaurants")
          .then()
             .statusCode(200)
             .extract()
             .asString();
    	
    	Approvals.verifyJson(response);
    }

}