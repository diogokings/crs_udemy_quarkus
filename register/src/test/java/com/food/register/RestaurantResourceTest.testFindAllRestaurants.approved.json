[
  {
    "name": "TEST_1",
    "owner": "TEST_1"
  },
  {
    "name": "TEST_2",
    "owner": "TEST_2"
  }
]